File: julia/src/BackendCUDA.jl
# File: Gain.jl
# Project: src
# Created Date: 14/06/2022
# Author: Shun Suzuki
# -----
# Last Modified: 14/06/2022
# Modified By: Shun Suzuki (suzuki@hapis.k.u-tokyo.ac.jp)
# -----
# Copyright (c) 2022 Shun Suzuki. All rights reserved.
# 

mutable struct BackendCUDA
    _backend_ptr::Ptr{Cvoid}
    function BackendCUDA()
        chandle = Ref(Ptr{Cvoid}(0))
        autd3capi_backend_cuda.autdcuda_backend(chandle)
        g = new(chandle[])
        finalizer(g -> autd3capi_gain_holo.autd_delete_backend(g._backend_ptr), g)
        g
    end
end

File: matlab/autd3/BackendCUDA.m
%{
%File: BackendCUDA.m
%Project: autd3
%Created Date: 10/06/2022
%Author: Shun Suzuki
%-----
%Last Modified: 10/06/2022
%Modified By: Shun Suzuki (suzuki@hapis.k.u-tokyo.ac.jp)
%-----
%Copyright (c) 2022 Shun Suzuki. All rights reserved.
%
%}

classdef BackendCUDA < Backend

    methods

        function obj = BackendCUDA()
            obj = obj@Backend();
            pp = libpointer('voidPtrPtr', obj.ptr);
            calllib('autd3capi_backend_cuda', 'AUTDCUDABackend', pp);
        end

    end

end
